app/src/main/java/com/github/godspeed010/weblib/fragments/WebViewFragment.kt

```diff
@@ -88,21 +102,53 @@ class WebViewFragment : Fragment() {
 
                 Log.d(TAG, "URL CHANGE to $url")
 
-                if (view != null) {
-                    //Update address bar
-                    view.findViewById<EditText>(R.id.et_address_bar).setText(url)
+                //Update address bar
+                view.apply {
+                    findViewById<EditText>(R.id.et_address_bar).setText(url)
+                }
+                pageError = false
 
-                    if (url != null) {
-                        lastVisitedUrl = url
+                if (url != null && lastVisitedUrl != url) {
+                    lastProgression = 0f
+                    lastVisitedUrl = url
+                }
+            }
+
+            override fun onReceivedError(wv: WebView?, request: WebResourceRequest?, error: WebResourceError?) {
+                super.onReceivedError(wv, request, error)
+
+                if (request?.url.toString() == wv?.url)
+                {
+                    Log.d(TAG, "An error occurred while loading page, disabling scroll.")
+                    pageError = true
+                }
+            }
+
+            override fun onPageFinished(wv: WebView?, loadedUrl: String?) {
+                super.onPageFinished(wv, loadedUrl)
+
+                if ((loadedUrl == novel.url) && !pageError && lastProgression != 0f) {
+                    scrollWebView(lastProgression, wv)
+                } else {
+                    view.apply {
+                        val progressBar = findViewById<View>(R.id.progressView)
+                        progressBar.visibility = View.GONE
                     }
```

Comment:
Similar to last comment, you're only using the context from the `apply` block in a single place, which is on line 134.

Additionally, `apply` shouldn't be used here because you're not modifying the view object. I shared that quote and link in the last review about when to use `apply` and the other blocks. You can take a look at that again.

```suggestion
                    val progressBar = view.findViewById<View>(R.id.progressView)
                    progressBar.visibility = View.GONE
```